src/Features/Core/Portable/Diagnostics/DefaultDiagnosticAnalyzerService.cs

```diff
@@ -229,9 +231,9 @@ public void RemoveProject(ProjectId projectId)
 
             private class DefaultUpdateArgsId : BuildToolId.Base<int, DocumentId>, ISupportLiveUpdate
             {
-                private readonly string _workspaceKind;
+                private readonly string? _workspaceKind;
 
-                public DefaultUpdateArgsId(string workspaceKind, AnalysisKind kind, DocumentId documentId) : base((int)kind, documentId)
+                public DefaultUpdateArgsId(string? workspaceKind, AnalysisKind kind, DocumentId documentId) : base((int)kind, documentId)
```

Comment:
Noticed `AnalysisKind` is declared in both Feature and Workspace. Probably should move the one in Feature to `Microsoft.CodeAnalysis.Diagnostics.EngineV2` namespace?

Reply:
Seems out of scope for this change.